nomad/state/schema.go

```diff
@@ -91,6 +91,15 @@ func jobTableSchema() *memdb.TableSchema {
 					Lowercase: true,
 				},
 			},
+			"type": &memdb.IndexSchema{
+				Name:         "type",
+				AllowMissing: false,
+				Unique:       false,
+				Indexer: &memdb.StringFieldIndex{
+					Field:     "Type",
+					Lowercase: false,
```

Comment:
The reason I didn't is that when you look-up a scheduler it is just looking up in a map where the keys are lower-case: https://github.com/hashicorp/nomad/blob/master/scheduler/scheduler.go#L23

So if you are not lower-case you wouldn't even be able to pick a scheduler. I think the correct fix is to lower-case user input when parsing the job file.